src/textual/app.py

```diff
@@ -2181,7 +2181,7 @@ async def push_screen_wait(
             The screen's result.
         """
         await self._flush_next_callbacks()
-        return await self.push_screen(screen, wait_for_dismiss=True)
+        return await asyncio.shield(self.push_screen(screen, wait_for_dismiss=True))
```

Comment:
Maybe a comment on _why_ we're shielding here/the circumstances that the shield protects against?